### ISSUE TITLE
Game crashes trying to render any sign

### ISSUE THREAD

**Issue by RegularPerson** [Team-Hibiscus/NatureSpirit#99]
I've been trying to troubleshoot this problem for a while now and I'm a bit stumped even. The main problem I run into is this NullPointerException crash whenever I place a Paper Sign or Hanging Paper Sign:
```
java.lang.NullPointerException: Cannot invoke "net.minecraft.class_4730.method_24145(net.minecraft.class_4597, java.util.function.Function)" because "$$7" is null
	at net.minecraft.class_837.method_45800(class_837.java:101)
	at net.minecraft.class_837.method_49920(class_837.java:81)
	at net.minecraft.class_7761.method_23083(class_7761.java:77)
	at net.minecraft.class_7761.method_3569(class_7761.java:30)
	at net.minecraft.class_824.method_23079(class_824.java:90)
	at net.minecraft.class_824.method_23081(class_824.java:79)
	at net.minecraft.class_824.method_3554(class_824.java:104)
	at net.minecraft.class_824.method_3555(class_824.java:79)
	at net.minecraft.class_761.method_22710(class_761.java:1367)
	at net.minecraft.class_757.method_3188(class_757.java:1110)
	at net.minecraft.class_757.method_3192(class_757.java:880)
	at net.minecraft.class_310.method_1523(class_310.java:1219)
	at net.minecraft.class_310.method_1514(class_310.java:802)
	at net.minecraft.client.main.Main.main(Main.java:250)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
```
Running a minecraft server with this mod on it too and other players run into this same problem and it results in us having to roll the server back to remove that sign.

**Comment by Eko-byte** [comment]
what mods are you using

**Comment by RegularPerson** [comment]
It's a modpack, so I can't exactly tell what would cause it. Here's the list:
```
        - amendments 1.20-1.2.12
	- another_furniture 1.20.1-3.0.1
	- architectury 9.2.14
	- balm-fabric 7.3.9
	- bbb 1.20.1-fabric-1.0.2
	- bounce_styles 2.5.1
	- cardinal-components 5.2.2
	- carryon 2.1.2.7
	- chatanimation 1.0.5
	- chimes 2.0.1
	- cloth-config 11.1.136
	- cobbled-shiny-particles 1.0.6.1
	- cobbledex 1.1.0
	- cobblefoods 1.3.2-1.20.1
	- cobblemon 1.5.2+1.20.1
	- cobblemon_capturexp 1.5-fabric-1.0.1
	- cobblemon_chaining 1.5-fabric-1.0.1
	- cobblemon_counter 1.5-fabric-1.2.0
	- cobblemon_fossiltweaks 1.5-fabric-1.0.0
	- cobblemon_move_inspector 1.1.1
	- cobblemon_spawn_notification 1.5-fabric-1.2.1
	- cobblemon_unimplementeditems 1.5-fabric-1.2.0
	- cobblemonbreakouts fabric-1.1.4-1.20.1
	- cobblemonchallenge 1.1.8
	- cobblemonridingfabric 1.2.6
	- cobblenav 1.3.5
	- cobbreeding 1.7.3
	- cobweb 1.0.0
	- collective 7.87
	- continuity 3.0.0-beta.5+1.20.1
	- controlling 12.0.2
	- crawl 0.12.0
	- easy_npc 5.7.6
	- emi 1.1.18+1.20.1+fabric
	- emojiful 4.2.0
	- entityculling 1.7.1
	- fabric-api 0.92.2+1.20.1
	- fabric-language-kotlin 1.12.3+kotlin.2.0.21
	- fabricloader 0.16.5
	- forgeconfigapiport 8.0.1
	- ftblibrary 2001.2.5
	- ftbteams 2001.3.0
	- geckolib 4.4.9
	- gymbadges 1.0
	- immersive_paintings 0.6.7+1.20.1
	- indium 1.0.34+mc1.20.1
	- inventoryprofilesnext 1.10.11
	- iris 1.7.5+mc1.20.1
	- item_obliterator 2.3.0
	- jade 11.11.0+fabric
	- java 17
	- jei 15.20.0.105
	- jumpoverfences 1.0-SNAPSHOT
	- justhammers 2.0.4+mc1.20.1
	- konkrete 1.8.1
	- lambdynlights 2.3.2+1.20.1
	- libipn 4.0.2
	- lithium 0.11.2
	- lootballs 1.1.4
	- luphieclutteredmod 2.1.0+1.20.1
	- mcwbridges 3.0.0
	- mcwdoors 1.1.1
	- mcwfences 1.1.2
	- mcwfurnitures 3.3.0
	- mcwlights 1.1.1
	- mcwpaths 1.0.5
	- mcwroofs 2.3.1
	- mcwstairs 1.0.0
	- mcwtrpdoors 1.1.3
	- mcwwindows 2.3.0
	- megamons 1.5.0+1.20.1-forge+fabric-75b66fa
	- melody 1.0.3
	- mermod 3.0.0
	- minecraft 1.20.1
	- modmenu 7.2.2
	- moonlight 1.20-2.13.28
	- more_cobblemon_tweaks 0.7.1
	- mousetweaks 2.26
	- myloot 4.0.3-1.20.1
	- mythsandlegends 1.6.1
	- natures_spirit 1.6.3-1.20.1
	- naturescompass 1.20.1-2.2.3-fabric
	- necronomicon 1.6.0
	- nightlights 1.1
	- numismatic-overhaul 0.2.14+1.20
	- openpartiesandclaims 0.23.2
	- owo 0.11.2+1.20
	- paperdoll 1.2.5
	- patchouli 1.20.1-84-FABRIC
	- pehkui 3.8.3+1.14.4-1.21
	- physicsmod 3.0.14
	- playerladder 0.6.0-1.20
	- pokeblocks 1.3.0-1.20.1
	- screenshot_viewer 1.3.1-fabric-mc1.20
	- seamless 2.2.2
	- searchables 1.0.3
	- simpletms 1.1.2
	- sit 1.20.1-27
	- slide_show 1.0.3-Enhancement-1
	- sodium 0.5.11+mc1.20.1
	- spoornpacks 5.0-1.20.1
	- terrablender 3.0.1.7
	- toms_storage 1.6.8
	- trashslot 15.1.0
	- travelersbackpack 9.1.18
	- travelerstitles 1.20-Fabric-4.0.2
	- treeharvester 9.1
	- trinkets 3.7.2
	- veinmining 1.5.0+1.20.1
	- voicechat 1.20.1-2.5.26
	- waystones 14.1.6
	- wildfire_gender 1.20-3.0.1
	- xaerominimap 24.6.1
	- xaeroworldmap 1.39.0
	- yet_another_config_lib_v3 3.6.1+1.20.1-fabric
	- yungsapi 1.20-Fabric-4.0.6
	- yuushya 2.2.2
	- yuushya_modelling 2.2.0
	- zoomify 2.14.2+1.20.1
```

**Comment by RegularPerson** [comment]
I realized that the same crash happens when placing _any_ sign so I've changed the title.

**Comment by Eko-byte** [comment]
Hey, I found the issue and I've fixed it on 2.0-1.21.1, when I get around to backporting, I will have this fixed, this only usually happens when certain mods are loaded but im not sure which ones from ur listt

**Comment by RegularPerson** [comment]
Thank you for looking into it either way! Depending on if we plan to update the server to 1.21 by then, I'll await the backport.
